module/actor/gurps-actor.ts

```diff
@@ -0,0 +1,656 @@
+    ;(this.system as BaseActorModel).prepareEmbeddedDocuments()
+  }
+
+  /* ---------------------------------------- */
+  /*  Legacy Functionality                    */
+  /* ---------------------------------------- */
+
+  async internalUpdate(data: Actor.UpdateData | undefined, operation?: Actor.Database.UpdateOperation) {
+    console.trace('internalUpdate', data)
+    return this.update(data, { ...operation, render: false })
+  }
+
+  /* ---------------------------------------- */
+
+  async addTaggedRollModifiers(
+    chatThing: string,
+    optionalArgs: { obj?: AnyObject },
+    attack?: MeleeAttackModel | RangedAttackModel
+  ): Promise<boolean> {
+    if (this.isOfType('character', 'enemy')) {
+      return this.system.addTaggedRollModifiers(chatThing, optionalArgs, attack)
+    }
+    console.warn('Actor is not a character or enemy.')
+    return false
+  }
+
+  /* ---------------------------------------- */
+
+  /**
+   * Parse roll info based on action type.
+   *
+   * @param {object} action - Object from GURPS.parselink
+   * @param {string} chatthing - internal code for roll
+   * @param {string} formula - formula for roll
+   * @param {string} thing - name of the source of the roll
+   * @returns {{}} result
+   * @returns {string} result.name - Name of the action which originates the roll
+   * @returns {[string]} result.uuid - UUID of the actor component that originates the roll
+   * @returns {[string]} result.itemId - ID of the item that originates the roll
+   * @returns {[string]} result.fromItem - ID of the parent item of the item that originates the roll
+   * @returns {[string]} result.pageRef - Page reference of the item that originates the roll
+   */
+  findUsingAction(
+    action: { type: string; name: string; orig: string },
+    chatthing: string,
+    formula: string,
+    thing: string
+  ): { name: string; uuid: string | null; itemId: string | null; fromItem: string | null; pageRef: string | null } {
+    if (this.isOfType('character', 'enemy')) {
+      return (this.system as CharacterModel).findUsingAction(action, chatthing, formula, thing)
+    }
+
+    console.warn('Actor is not a character or enemy.')
+    return {
+      name: thing,
+      uuid: null,
+      itemId: null,
+      fromItem: null,
+      pageRef: null,
+    }
+  }
+
+  /* ---------------------------------------- */
+
+  // TODO: review and refactor
+  async changeDR(
+    formula: string,
+    locations: string[]
+  ): Promise<{ changed: boolean; msg: string; warn?: string; info?: string }> {
+    if (this.isOfType('character', 'enemy')) {
+      return this.system.changeDR(formula, locations)
+    }
+    return { changed: false, msg: '', warn: 'Actor is not a character or enemy.' }
+  }
+
+  /* ---------------------------------------- */
+
+  // Check if a roll can be performed.
+  // NOTE: there doesn't seem to be much reason for this method to be in the Actor class.
+  // Consider moving it to roll or elsewhere.
```

Comment:
I may disagree with the above comment. The `canRoll` function uses data from the actor, such as `maneuver`, `extraAttacks`, `rapidStrikeBonus`, `maxBlocks`, etc. That leads me towards putting it on Actor.